### PR TITLE
Worn letters!

### PR DESCRIPTION
**Pull request by jasonmoonen**
There's a 10% chance to receive a worn letter while killing zombies by default. Rewards are currently static, except for a 50% chance of getting a shotgun or an AR-15.

### CODE REVIEW

File: src/servers/ZoneServer2016/entities/treasurechest.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                guid: this.characterId,
+                stringId: StringIds.OPEN
+            });
+        }
+    }
+    /* eslint-disable @typescript-eslint/no-unused-vars */
+    OnPlayerSelect(
+        server: ZoneServer2016,
+        client: ZoneClient2016,
+        isInstant?: boolean
```
Comment (QuentinGruber) [inline src/servers/ZoneServer2016/entities/treasurechest.ts:112]: Don't let unused vars :cowboy_hat_face:

File: src/servers/ZoneServer2016/managers/worldobjectmanager.ts [hunk 2]
```diff
@@ -342,6 +345,28 @@ export class WorldObjectManager {
   }
 
   createLootbag(server: ZoneServer2016, entity: BaseFullCharacter) {
+    if (entity instanceof Zombie) {
+      //TODO: Probably should rework this?
+      const wornLetters = [
```
Comment (QuentinGruber) [inline src/servers/ZoneServer2016/managers/worldobjectmanager.ts:350]: wornLetters could be a static var of worlObjectManager

**Review by QuentinGruber — COMMENTED** [review]
Thx that's awesome !

**Comment by QuentinGruber** [comment]
I've tested it ingame and it doesn't work for me

**Comment by jasonmoonen** [comment]
> I've tested it ingame and it doesn't work for me

Did you update the config file? I’ve had this happen while adding the config for water sources.

if the config is not updated the variable will be undefined by default, so it won’t give rewards.

Maybe it’s a good idea to implement a config versioning and upgrade the older config acordingly?

**Comment by QuentinGruber** [comment]
> > I've tested it ingame and it doesn't work for me
> 
> Did you update the config file? I’ve had this happen while adding the config for water sources.
> 
> if the config is not updated the variable will be undefined by default, so it won’t give rewards.
> 
> Maybe it’s a good idea to implement a config versioning and upgrade the older config acordingly?

Ok now i see them but can't interact with the lootbag, and config versioning is kinda already a thing.

**Comment by jasonmoonen** [comment]
> > > I've tested it ingame and it doesn't work for me
> > 
> > 
> > Did you update the config file? I’ve had this happen while adding the config for water sources.
> > if the config is not updated the variable will be undefined by default, so it won’t give rewards.
> > Maybe it’s a good idea to implement a config versioning and upgrade the older config acordingly?
> 
> Ok now i see them but can't interact with the lootbag, and config versioning is kinda already a thing.

Hey @QuentinGruber You should be able to take the worn letters now

**Comment by QuentinGruber** [comment]
> > > > I've tested it ingame and it doesn't work for me
> > > 
> > > 
> > > Did you update the config file? I’ve had this happen while adding the config for water sources.
> > > if the config is not updated the variable will be undefined by default, so it won’t give rewards.
> > > Maybe it’s a good idea to implement a config versioning and upgrade the older config acordingly?
> > 
> > 
> > Ok now i see them but can't interact with the lootbag, and config versioning is kinda already a thing.
> 
> Hey @QuentinGruber You should be able to take the worn letters now

yep it works now !

**Pull request closed without merging**